modules/flowable-app-engine/src/main/java/org/flowable/app/engine/AppEngineConfiguration.java

```diff
@@ -393,7 +393,7 @@ public void initVariableTypes() {
             }
             variableTypes.addType(new NullType());
             variableTypes.addType(new StringType(getMaxLengthString()));
```

Comment:
@filiphr What if (getMaxLengthString() > maxAllowedLengthVariableType) ?

Reply:
That's an edge case that people configuring should take into consideration. The main use case for this new property is for way larger variables, not the typical `maxLengthString` of 2K or 4K characters. 

I'll see if I can do something about that